implemented file server

FaiyazMujawar opened the pull request:
Fixes #1 

Created a basic file server for saving images.
commit

The data should be sent to `{base_uri}/image` in `multipart/form-data` format. The data should be sent as follows:
```JSON
{
    "userId":"ID of the user",
    "images":"images, if any"
}
```

PS: the order of fields must be same.

drigio commented on index.js:
```diff
@@ -0,0 +1,14 @@
+// Use 'public' folder for serving images
+app.use("/public/images", express.static("public"));
+
+app.use("/", require("./routes/file"));
```
`/image` route might be better.

FaiyazMujawar replied:
Okay

drigio commented on routes/file.js:
```diff
@@ -0,0 +1,15 @@
+const router = require("express").Router();
+const saveFile = require("../services/MulterService");
```
`saveImage` might be a better name. In the future we might have different types of media as well.

FaiyazMujawar replied:
I was thinking of making it generalized. I can change if you say so.

drigio commented on services/MulterService.js:
```diff
@@ -0,0 +1,18 @@
+const path = require("path");
+
+/** Maximum image count */
+const maxImages = 3;
```
Capitalize constants.

drigio commented on services/MulterService.js:
```diff
@@ -0,0 +1,18 @@
+const maxImages = 3;
+
+const storage = multer.diskStorage({
+  destination: "./public/images",
```
make destination a constant as well.

drigio reviewed (changes requested):
Added a few comments. Please have a look.

`multerImageService.js` would be an appropriate name.

FaiyazMujawar committed:
implemented file server

Created a basic file server for saving images.
commit

The data should be sent to `{base_uri}/image` in `multipart/form-data` format. The data should be sent as follows:
```JSON
{
    "userId":"ID of the user",
    "images":"images, if any"
}
```

PS: the order of fields must be same.

drigio commented:
LGTM! Pulling in changes.
